### PR TITLE
Un-Authorized and Un-Authenticated unit tests added

### LINKED ISSUE #29: Add unit tests for Unauthorised method calls.

### PR DESCRIPTION
**Pull request by noveens**
Fixes #29 

Added unauthorized and unauthenticated tests.

@PVince81

### CODE REVIEW

File: owncloud/helperFunctions.js [hunk 1]
```diff
@@ -641,6 +675,26 @@ helpers.prototype._OCSuserResponseHandler = function(data, resolve, reject) {
  */
 helpers.prototype._getAllFileInfo = function(path, pathToStore) {
 	function getAllFileInfo(path, pathToStore, localPath) {
+		var fl = 0;
+		var baseAddr = pathToStore;
+
+		for (var j=0;j<filesToPut.length;j++) {
+			if (filesToPut[j].path === baseAddr) {
```
Comment (PVince81) [inline owncloud/helperFunctions.js]: where is `filesToPut` defined ? I'd expect jshint to complain

File: owncloud/helperFunctions.js [hunk 2]
```diff
@@ -667,12 +721,12 @@ helpers.prototype._getAllFileInfo = function(path, pathToStore) {
...
 
 				if (fl === 0) {
-					var count = filesToPut.length;
-					filesToPut[count] = {};
-					filesToPut[count].path = baseAddr;
-					filesToPut[count].localPath = localPath; ////////
-					filesToPut[count].files = [ file ];
-					count++;
+					var count2 = filesToPut.length;
+					filesToPut[count2] = {};
```
Comment (PVince81) [inline owncloud/helperFunctions.js]: are you appending values to the `filesToPut` array this way ?
Usually using `filesToPut.push(value)` is better.

File: owncloud/helperFunctions.js (hunk 1 above)
Comment (noveens) [inline owncloud/helperFunctions.js]: Here: https://github.com/noveens/js-owncloud-client/pull/35/files/0ce18f736d53a97eec56e1e8aa4afbf56c9f73e0#diff-b9b823e719730e39048dffbb512e743cR735

File: owncloud/helperFunctions.js (hunk 2 above)
Comment (noveens) [inline owncloud/helperFunctions.js]: I guess this should be fine anyway,
Should I send in a PR?

**Pull request merged**